packages/components/src/internal/components/search/EntityFieldFilterModal.tsx

```diff
@@ -188,14 +211,27 @@ export const EntityFieldFilterModal: FC<Props> = memo(props => {
         Object.keys(dataTypeFilters).forEach(parent => {
             const filterFields = dataTypeFilters[parent];
             filterFields.forEach(field => {
-                const key = parent + '-' + field.filter.getColumnName();
+                const key = parent + '-' + field.fieldKey;
                 status[key] = true;
             });
         });
 
         return status;
     }, [dataTypeFilters]);
 
+    const fieldDistinctValueFilters = useMemo(() => {
+        if (!dataTypeFilters || !activeQuery || !activeField) return null;
+
+        const filters = [];
+
+        // use active filters to filter distinct values, but exclude filters on current field
+        dataTypeFilters?.[activeQuery]?.forEach(field => {
+            if (field.fieldKey !== activeField.fieldKey) filters.push(field.filter);
+        });
```

Comment:
could this be simplified with using `.filter()` instead of `.forEach()`?

Reply:
It's a little different from filter, because the new array just need the field.filter value, not the whole field value.